python/src/specifications/specifications_impl.py

```diff
@@ -22,40 +34,39 @@ def spec_csv(architecture_file: str, output_directory: str) -> PluginExecutionRe
         output_directory (str): The directory to write csv spec content.
     """
 
-    def get_csv():
-        spec_definitions = _get_parsed_models(architecture_file)
-        reqs = {}
-        for spec in spec_definitions:
-            file_name = path.basename(spec.source.uri)
-            reqs[spec.name] = _get_requirements(spec)
+    spec_definitions = parse(architecture_file)
+    reqs = {}
+    for spec in spec_definitions:
+        file_name = path.basename(spec.source.uri)
+        reqs[spec.name] = _get_requirements(spec)
 
-        field_names = ["Spec Name", "Section", "ID", "Requirement", "Parents", "Children"]
+    field_names = ["Spec Name", "Section", "ID", "Requirement", "Parents", "Children"]
 
-        # just in case, let's make sure the output directory exists
-        if not path.lexists(output_directory):
-            makedirs(output_directory)
+    # just in case, let's make sure the output directory exists
+    if not path.lexists(output_directory):
+        makedirs(output_directory)
 
-        file_counter = 0
-        for spec_name in reqs.keys():
-            file_name = spec_name + ".csv"
-            file_name = file_name.replace(" ", "_")
-            output_path = path.join(output_directory, file_name)
-            with open(output_path, "w") as output:
-                writer = csv.DictWriter(output, fieldnames=field_names)
-                writer.writeheader()
-                writer.writerows(reqs[spec_name])
-                file_counter = file_counter + 1
+    file_counter = 0
+    for spec_name in reqs.keys():
+        file_name = spec_name + ".csv"
+        file_name = file_name.replace(" ", "_")
+        output_path = path.join(output_directory, file_name)
+        with open(output_path, "w") as output:
+            writer = csv.DictWriter(output, fieldnames=field_names)
+            writer.writeheader()
+            writer.writerows(reqs[spec_name])
+            file_counter = file_counter + 1
 
-        return f"{file_counter} CSV spec files written to {output_directory}"
+    status = ExecutionStatus.SUCCESS
+    messages: list[ExecutionMessage] = []
+    messages.append(ExecutionMessage(f"{file_counter} CSV spec files written to {output_directory}", MessageLevel.INFO, None, None))
 
-    with plugin_result(plugin_name, get_csv) as result:
-        return result
+    return ExecutionResult(plugin_name, "Specifications", status, messages)
 
 
-def _get_parsed_models(architecture_file: str) -> List[Definition]:
-    with validated_source(architecture_file) as result:
-        return result.definitions
-
+# def _get_parsed_models(architecture_file: str) -> List[Definition]:
+#     with validated_source(architecture_file) as result:
+#         return result.definitions
 
 def _get_requirements(spec: Definition) -> List[dict]:
     ret_val: List[dict] = []
```

Comment:
Unfortunately, I can't highlight after this line to help capture what I am wanting to reference. But there should have been additional changes take place within the remaining two methods' logic. 

So, while technically there is not anything wrong with this and it would still work this way because the `structure` attribute of `Definition` still exists and is a dictionary, so you can access the pieces contained within in the below manner. But part of the entire point of the 0.4.x baseline was to provide a Python object concept of the `Definition` to provide a way to access and reference the information of the definition. That is contained within the `instance` attribute of the `Definition`. 

So, you need to take a look at the plugins within core to get some of the data object transitions that should have taken place.

i.e. `spec_dict: dict = spec.structure` should be more like `spec_definition = spec.instance`
`spec_name = spec_dict["spec"]["name"]` should be more like `spec_name = spec_definition.name`

Take a look at `check`, `root schema must have name`, and `no extension for final` for more insight to some of the change overs.